plugins/cpu/cpu.go

```diff
@@ -13,31 +21,170 @@ const (
 	niceGauge
 )
 
+// Hold cpu sampled data and calculate moving averages
+type sample struct {
+	alpha        float64
+	values       []float64
+	expMovingAvg float64
+}
+
+func newSample(alpha float64) *sample {
+	return &sample{alpha: alpha}
+}
+
+func (s *sample) add(v float64) {
+	s.values = append(s.values, v)
+
+	if s.expMovingAvg == 0 {
+		s.expMovingAvg = v
+	} else {
+		s.expMovingAvg = v*s.alpha + (s.expMovingAvg * (1 - s.alpha))
+	}
+}
+
+// Reset data at reporting intervals
+func (s *sample) drain() float64 {
+	avg := s.expMovingAvg
+	s.expMovingAvg = 0
+	s.values = []float64{}
```

Comment:
And if you're doing moving averages using the samples, wouldn't it be better to use a circular buffer instead of resetting at reporting intervals?